src/app/lib/utilities/preprocessor/rules/cpsAssetPage/addSummaryBlock/index.js

```diff
@@ -0,0 +1,40 @@
+import pathOr from 'ramda/src/pathOr';
+import deepClone from 'ramda/src/clone';
+import path from 'ramda/src/path';
+import { singleTextBlock } from '#app/models/blocks';
+
+const getSummary = json => {
+  const summary = pathOr(null, ['promo', 'summary'], json);
+
+  if (!summary) {
+    return json;
+  }
+
+  return summary;
+};
+
+const addSummaryBlocks = originalJson => {
+  const json = deepClone(originalJson);
+  const pageType = path(['metadata', 'type'], json);
+  const blocks = path(['content', 'model', 'blocks'], json);
+
+  if (pageType !== 'PGL' || !blocks) {
+    return json;
+  }
+
+  const headline = blocks.find(({ type }) =>
+    ['headline', 'fauxHeadline'].includes(type),
+  );
```

Comment:
I wondered about this logic... Shouldn't the headline block already be first (from the preceding rule, `addHeadlineBlock`)? And AFAICS (in the same file)`fauxHeadline` is only added if it's a `MAP`, which it can't be at this point. Or can it come from somewhere else? Just pondering the potential for unexpected side-effects.

Reply:
My understanding of these rules is they work as a stream, each rule taking the full data json and modifying it as that rule sees fit and passing it on to the next one. That was my main reservation with these rules, they are all coupled to each other, they take what another rule has done and build on it. In this case the headline rule runs, and then this rule finds the headline and inserts the summary after it.

I think what you are imagining is a list that is added to after each rule, that would be much more intuitive and a structure I'd favour more. That was the list of 'rules' would reflect the order of the document.

Hope that makes sense, bit late in the day :/

Reply:
Does make sense, thanks. In that light I think we should go with this now and then start to design a better solution together that we want to move towards as otherwise this is going to get much more hairy once we get to STYs